app/src/main/java/com/babylon/wallet/android/presentation/settings/accountsecurity/ledgerhardwarewallets/LedgerHardwareWalletsViewModel.kt

```diff
@@ -65,10 +115,16 @@ class LedgerHardwareWalletsViewModel @Inject constructor(
 data class LedgerHardwareWalletsUiState(
     val loading: Boolean = false,
     val showContent: ShowContent = ShowContent.Details,
-    val ledgerDevices: ImmutableList<LedgerHardwareWalletFactorSource> = persistentListOf()
+    val ledgerDevices: ImmutableList<LedgerHardwareWalletFactorSource> = persistentListOf(),
+    val showLinkConnectorPromptState: ShowLinkConnectorPromptState = ShowLinkConnectorPromptState.None,
+    val isLinkConnectionEstablished: Boolean = false,
+    val linkingToConnector: Boolean = false
 ) : UiState {
 
-    enum class ShowContent {
-        Details, AddLedger, LinkNewConnector, AddLinkConnector
+    sealed interface ShowContent {
+        data object Details : ShowContent
+        data object AddLedger : ShowContent
+        data class LinkNewConnector(val addDeviceAfterLinking: Boolean = true) : ShowContent
+        data class AddLinkConnector(val addDeviceAfterLinking: Boolean = true) : ShowContent
```

Comment:
All the logic of this viewmodel can be written with the following state data class:

```
data class LedgerHardwareWalletsUiState(
    val loading: Boolean = false,
    val showContent: ShowContent = ShowContent.Details,
    val ledgerDevices: ImmutableList<LedgerHardwareWalletFactorSource> = persistentListOf(),
    val isAddLedgerButtonEnabled: Boolean = false 
) : UiState {

    sealed interface ShowContent {
        data object Details : ShowContent
        data object AddLedger : ShowContent
        data class LinkNewConnector(val showPrompt: Boolean = false) : ShowContent
        data object AddLinkConnector : ShowContent
    }
}
```

- the `addDeviceAfterLinking` is never used in the screen
- the `LinkNewConnector` state can be used to show or not the dialog
- the `linkingToConnector` can be renamed to `isAddLedgerButtonEnabled` since it is used to enable/disable the button

Reply:
this prompt is not displayed from `LinkNewConnector` screens, so I don't think we can simplify state as you wrote

Reply:
this prompt is not displayed from `LinkNewConnector` screens, so I don't think we can simplify state as you wrote